backend/dataall/modules/worksheets/services/worksheet_query_result_service.py

```diff
@@ -0,0 +1,121 @@
+import csv
+import io
+import os
+from datetime import datetime, timedelta
+from typing import TYPE_CHECKING
+
+from openpyxl import Workbook
+
+from dataall.base.db import exceptions
+from dataall.core.environment.services.environment_service import EnvironmentService
+from dataall.core.permissions.services.tenant_policy_service import TenantPolicyService
+from dataall.modules.worksheets.aws.s3_client import S3Client
+from dataall.modules.worksheets.db.worksheet_models import WorksheetQueryResult
+from dataall.modules.worksheets.db.worksheet_repositories import WorksheetRepository
+from dataall.modules.worksheets.services.worksheet_enums import WorksheetResultsFormat
+from dataall.modules.worksheets.services.worksheet_permissions import RUN_ATHENA_QUERY_TENANT
+from dataall.modules.worksheets.services.worksheet_service import WorksheetService
+
+if TYPE_CHECKING:
+    try:
+        from sqlalchemy.orm import Session
+        from openpyxl.worksheet.worksheet import Worksheet
+    except ImportError:
+        print('skipping type checks as stubs are not installed')
+        Session = None
+        Worksheet = None
+
+
+class WorksheetQueryResultService:
+    _DEFAULT_ATHENA_QUERIES_PATH = 'athenaqueries'
+    _DEFAULT_QUERY_RESULTS_TIMEOUT = os.getenv('QUERY_RESULT_TIMEOUT_MINUTES', 120)
+
+    @staticmethod
+    def _create_query_result(
+        environment_bucket: str, athena_workgroup: str, worksheet_uri: str, region: str, aws_account_id: str, data: dict
+    ) -> WorksheetQueryResult:
+        sql_query_result = WorksheetQueryResult(
+            worksheetUri=worksheet_uri,
+            AthenaQueryId=data.get('athenaQueryId'),
+            fileFormat=data.get('fileFormat'),
+            OutputLocation=f's3://{environment_bucket}/athenaqueries/{athena_workgroup}/',
```

Comment:
nit: use `_DEFAULT_ATHENA_QUERIES_PATH` instead of hardcoded `athenaqueries` in line 41

Reply:
done